l1-contracts-foundry/script/DeployL1.s.sol

```diff
@@ -0,0 +1,664 @@
+        console.log("Erc20BridgeProxy deployed at:", contractAddress);
+        addresses.bridges.erc20BridgeProxy = contractAddress;
+    }
+
+    function deploySharedBridgeContracts() internal {
+        deploySharedBridgeImplementation();
+        deploySharedBridgeProxy();
+        registerSharedBridge();
+    }
+
+    function deploySharedBridgeImplementation() internal {
+        bytes memory bytecode = abi.encodePacked(
+            type(L1SharedBridge).creationCode,
+            // solhint-disable-next-line func-named-parameters
+            abi.encode(
+                config.tokens.tokenWethAddress,
+                addresses.bridgehub.bridgehubProxy,
+                addresses.bridges.erc20BridgeProxy,
+                config.eraChainId,
+                addresses.bridges.erc20BridgeImplementation,
+                addresses.stateTransition.diamondProxy
+            )
+        );
+        address contractAddress = deployViaCreate2(bytecode);
+        console.log("SharedBridgeImplementation deployed at:", contractAddress);
+        addresses.bridges.sharedBridgeImplementation = contractAddress;
+    }
+
+    function deploySharedBridgeProxy() internal {
+        uint256 storageSwitch = config.contracts.sharedBridgeUpgradeStorageSwitch;
+        bytes memory initCalldata = abi.encodeCall(L1SharedBridge.initialize, (addresses.governance, storageSwitch));
+        bytes memory bytecode = abi.encodePacked(
+            type(TransparentUpgradeableProxy).creationCode,
+            abi.encode(addresses.bridges.sharedBridgeImplementation, addresses.transparentProxyAdmin, initCalldata)
+        );
+        address contractAddress = deployViaCreate2(bytecode);
+        console.log("SharedBridgeProxy deployed at:", contractAddress);
+        addresses.bridges.sharedBridgeProxy = contractAddress;
+    }
+
+    function registerSharedBridge() internal {
+        Bridgehub bridgehub = Bridgehub(addresses.bridgehub.bridgehubProxy);
+        vm.startBroadcast();
+        bridgehub.addToken(address(0x01));
+        bridgehub.setSharedBridge(addresses.bridges.sharedBridgeProxy);
+        vm.stopBroadcast();
+        console.log("SharedBridge registered");
+    }
+
+    function deployErc20BridgeImplementation() internal {
+        bytes memory bytecode = abi.encodePacked(
+            type(L1ERC20Bridge).creationCode,
+            abi.encode(addresses.bridges.sharedBridgeProxy)
+        );
+        address contractAddress = deployViaCreate2(bytecode);
+        console.log("Erc20BridgeImplementation deployed at:", contractAddress);
+        addresses.bridges.erc20BridgeImplementation = contractAddress;
+    }
+
+    function upgradeL1Erc20Bridge() internal {
+        // In local network, we need to change the block.number
+        // as the operation could be scheduled for timestamp 1
+        // which is also a magic number meaning the operation
+        // is done.
+        if (config.chainId == 31337) {
+            vm.warp(10);
+        }
```

Comment:
Hm, i'm not sure i follow it

Reply:
Basically given timestamp = 1 is a special number (https://github.com/matter-labs/era-contracts/blob/4aa7006153ad571643342dff22c16eaf4a70fdc1/l1-contracts/contracts/governance/Governance.sol#L22) if I don't change the block timestamp to say 10 (any number over 1 will work), then the scheduled operation (https://github.com/aon/era-contracts/blob/e1210af78136a944b73ee2fd06f3c72d22033591/l1-contracts-foundry/script/DeployL1.s.sol#L564) for the next block timestamp, which is 1 when running a local node, will think it's already executed and won't allow for execution.